src/main/java/appeng/container/implementations/ContainerFluidTerminal.java

```diff
@@ -0,0 +1,479 @@
+			}
+			if( stack != null && this.clientRequestedTargetFluid != null && stack.getFluidStack().isFluidEqual( this.clientRequestedTargetFluid.getFluidStack() ) )
+			{
+				return;
+			}
+			NetworkHandler.instance().sendToServer( new PacketTargetFluidStack( (AEFluidStack) stack ) );
+		}
+
+		this.clientRequestedTargetFluid = stack == null ? null : stack.copy();
+	}
+
+	@Override
+	public void updateSetting( IConfigManager manager, Enum settingName, Enum newValue )
+	{
+		if( this.getGui() != null )
+		{
+			this.getGui().updateSetting( manager, settingName, newValue );
+		}
+	}
+
+	@Override
+	public void detectAndSendChanges()
+	{
+		if( Platform.isServer() )
+		{
+			if( this.monitor != this.terminal.getInventory( AEApi.instance().storage().getStorageChannel( IFluidStorageChannel.class ) ) )
+			{
+				this.setValidContainer( false );
+			}
+
+			for( final Settings set : this.serverCM.getSettings() )
+			{
+				final Enum<?> sideLocal = this.serverCM.getSetting( set );
+				final Enum<?> sideRemote = this.clientCM.getSetting( set );
+
+				if( sideLocal != sideRemote )
+				{
+					this.clientCM.putSetting( set, sideLocal );
+					for( final IContainerListener crafter : this.listeners )
+					{
+						if( crafter instanceof EntityPlayerMP )
+						{
+							try
+							{
+								NetworkHandler.instance().sendTo( new PacketValueConfig( set.name(), sideLocal.name() ), (EntityPlayerMP) crafter );
+							}
+							catch( final IOException e )
+							{
+								AELog.debug( e );
+							}
+						}
+					}
+				}
+			}
+
+			if( !this.fluids.isEmpty() )
+			{
+				try
+				{
+					final IItemList<IAEFluidStack> monitorCache = this.monitor.getStorageList();
+
+					final PacketMEFluidInventoryUpdate piu = new PacketMEFluidInventoryUpdate();
+
+					for( final IAEFluidStack is : this.fluids )
+					{
+						final IAEFluidStack send = monitorCache.findPrecise( is );
+						if( send == null )
+						{
+							is.setStackSize( 0 );
+							piu.appendFluid( is );
+						}
+						else
+						{
+							piu.appendFluid( send );
+						}
+					}
+
+					if( !piu.isEmpty() )
+					{
+						this.fluids.resetStatus();
+
+						for( final Object c : this.listeners )
+						{
+							if( c instanceof EntityPlayer )
+							{
+								NetworkHandler.instance().sendTo( piu, (EntityPlayerMP) c );
+							}
+						}
+					}
+				}
+				catch( final IOException e )
+				{
+					AELog.debug( e );
+				}
+			}
+			this.updatePowerStatus();
+
+			super.detectAndSendChanges();
+		}
+	}
+
+	protected void updatePowerStatus()
+	{
+		try
+		{
+			if( this.networkNode != null )
+			{
+				this.setPowered( this.networkNode.isActive() );
+			}
+			else if( this.getPowerSource() instanceof IEnergyGrid )
+			{
+				this.setPowered( ( (IEnergyGrid) this.getPowerSource() ).isNetworkPowered() );
+			}
+			else
+			{
+				this.setPowered( this.getPowerSource().extractAEPower( 1, Actionable.SIMULATE, PowerMultiplier.CONFIG ) > 0.8 );
+			}
+		}
+		catch( final Throwable t )

```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1166)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Catch Exception instead of Throwable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1181)